src/bitcoin/fee.rs

```diff
@@ -222,18 +219,26 @@ impl Fee for PartiallySignedTransaction {
 
         let input_sum = get_available_input_sat(self)?.as_sat();
         let output_sum = self.unsigned_tx.output[0].value;
-        let fee = input_sum
+        let effective_fee = input_sum
             .checked_sub(output_sum)
             .ok_or(FeeStrategyError::AmountOfFeeTooHigh)?;
-        let weight = self.unsigned_tx.weight() as u64;
-
-        let effective_sat_per_kvb = SatPerKvB::from_sat(
-            (weight * 1000)
-                .checked_div(fee)
-                .ok_or(FeeStrategyError::AmountOfFeeTooLow)?,
-        );
 
-        Ok(strategy.check(&effective_sat_per_kvb))
+        // simulate witness data
+        let mut tx = self.unsigned_tx.clone();
+        tx.input[0].witness = Witness::from_vec(vec![vec![0; 71], vec![0; 72], vec![0; 70]]);
```

Comment:
I'm guessing here: Is this supposed to simulate 3 signatures as an upper bound for witness that consist of two signatures and some op codes?

Reply:
352c57ba275565e1c5077d374c60c5b8ddeee310